mysql-test/main/max_used_connections_time.test

```diff
@@ -0,0 +1,35 @@
+-- source include/not_embedded.inc
+
+--echo #
+--echo # MDEV-30543 New status variable: Max_used_connections_time
+--echo #
+
+FLUSH STATUS;
+--disable_query_log
+--disable_result_log
+connect (con1,localhost,root,,);
+--sleep 1
+connect (con2,localhost,root,,);
+--sleep 1
+let $time_1=`SELECT VARIABLE_VALUE FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'max_used_connections_time'`;
+--sleep 1
+connection con1;
+disconnect con2;
+--sleep 1
+let $time_2=`SELECT VARIABLE_VALUE FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'max_used_connections_time'`;
+--echo # This should report 0 as it must have the same value as above
+let $dif_1= `SELECT '$time_1' <> '$time_2'`;
+--echo $dif_1
+--sleep 1
+connect (con3,localhost,root,,);
+--sleep 1
+connect (con4,localhost,root,,);
+--sleep 1
+let $time_3=`SELECT VARIABLE_VALUE FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME LIKE 'max_used_connections_time'`;
+--echo # This should report 1 as connection number was increased
```

Comment:
Missing test case of setting the value and generating the error message.

Reply:
`max_used_connections_time` appears in `SHOW STATUS` output; it is not a system variable so there is no SQL-level mechanism to even attempt to set it.

Are you sure this is relevant?

Reply:
You mentioned [`main/status.test`](https://github.com/MariaDB/server/blob/e416fb850bf06399201d59a5c711b7ff754e1d2f/mysql-test/main/status.test#L423-L431), where there are similarly no attempts to `SET` the values of the "status variables".

Reply:
I don't think I understand what you're asking for here.

This PR adds _one_ status variable, `max_used_connections_time`, with one clear purpose and simple implementation: "record the timestamp at which `max_used_connections` was last set."

@Chaloff has correspondingly added one test, which functionally verifies the behavior of this status variable.

Reply:
There's no added error message. I don't understand either. I think the test case looks fine, though moving it with other test cases would be good.